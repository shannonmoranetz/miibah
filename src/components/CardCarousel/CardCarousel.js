import React, { Component } from 'react';
import { connect } from 'react-redux';
import Card from '../../containers/Card/Card.js';

export class CardCarousel extends Component {

  returnAmiibos = (amiibos) => {
      return amiibos.map((amiibo, i) => {
      return <Card name = {amiibo.name}
                   amiiboSeries = {amiibo.amiiboSeries}
                   gameSeries = {amiibo.gameSeries}
                   image = {amiibo.image}
                   release = {amiibo.release.na}
                   id = {amiibo.id}
                   key = {i} />
    })
  }

  render() {
    console.log(this.props)
    return (
      <div className="CardCarousel">
        { this.props.match.path === '/' && this.returnAmiibos(this.props.amiibos) }
        { this.props.match.path === '/amiibo/:id' && this.returnAmiibos(this.props.amiibos) }
      </div>
    )
  }
}

export const mapStateToProps = state => ({
  amiibos: state.amiibos
});

export default connect(mapStateToProps)(CardCarousel);
