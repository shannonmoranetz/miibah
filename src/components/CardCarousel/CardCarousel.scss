.CardCarousel {
  display: flex;
  overflow: scroll;
  overflow-y: hidden;
  background-color: $brightpink;
  margin: 20px 0px;
  border-top: 20px solid $darkblue;
  border-bottom: 20px solid $darkblue;
}