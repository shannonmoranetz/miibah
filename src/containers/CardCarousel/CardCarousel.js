import React, { Component } from 'react';
import { connect } from 'react-redux';
import Card from '../../components/Card/Card.js';

export class CardCarousel extends Component {

  returnAmiibos = (amiibos) => {
      return amiibos.map((amiibo, i) => {
        return <Card  name = {amiibo.name}
                      amiiboSeries = {amiibo.amiiboSeries}
                      gameSeries = {amiibo.gameSeries}
                      image = {amiibo.image}
                      release = {amiibo.release.na}
                      key = {i} />
    })
  }

  render() {
    return (
      <div className="CardCarousel">
        { this.props.match.path === '/' && this.returnAmiibos(this.props.amiibos) }
        { this.props.match.path === '/wishlist' && this.returnAmiibos(this.props.wishlist) }
        { this.props.match.path === '/collected' && this.returnAmiibos(this.props.collected) }
        { this.props.match.path === '/amiibo/:name' && this.returnAmiibos(this.props.amiibos) }
        { this.props.match.path === '/search/:name' && this.returnAmiibos(this.props.searchedAmiibo) }
      </div>
    )
  }
}

export const mapStateToProps = state => ({
  amiibos: state.amiibos,
  searchedAmiibo: state.searchedAmiibo,
  wishlist: state.wishlist,
  collected: state.collected
});

export default connect(mapStateToProps)(CardCarousel);
