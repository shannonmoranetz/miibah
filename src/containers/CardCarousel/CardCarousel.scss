.CardCarousel {
  display: flex;
  overflow: scroll;
  overflow-y: hidden;
  background-color: $brightpink;
  margin: 20px 0px;
  border-top: 20px solid $darkblue;
  border-bottom: 20px solid $darkblue;
}

.empty-message {
  font-family: $heading-font;
  color: white;
  text-align: center;
  font-size: 30px;
}