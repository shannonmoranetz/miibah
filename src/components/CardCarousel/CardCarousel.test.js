import React from 'react';
import CardCarousel, { mapStateToProps } from './CardCarousel';
import { shallow } from 'enzyme';

describe('CardCarousel', () => {
  let wrapper;
  const amiibos = [{id: 1}, {id: 2}]
  beforeEach(() => {
    wrapper = shallow(<CardCarousel />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('mapStateToProps', () => {
    it('should return an object with an amiibos array', () => {
      const expected = amiibos;
      const mockState = { amiibos: [{id: 1}, {id: 2}] }
      const props = mapStateToProps(mockState);
      expect(props.amiibos).toEqual(expected);
    });
    
  });
});