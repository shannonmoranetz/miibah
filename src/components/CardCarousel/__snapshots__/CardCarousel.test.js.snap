// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardCarousel should match the snapshot 1`] = `undefined`;
